wasm/jit/jit_amd64_test.go

```diff
@@ -1342,17 +1336,267 @@ func TestAmd64Builder_handleLe(t *testing.T) {
+				require.NotContains(t, compiler.locationStack.usedRegisters, x2.register)
+				// To verify the behavior, we push the flag value
+				// to the stack.
+				top := compiler.locationStack.peek()
+				require.True(t, top.onConditionalRegister() && !top.onRegister())
+				err = compiler.moveConditionalToGeneralPurposeRegister(top)
+				require.NoError(t, err)
+				require.True(t, !top.onConditionalRegister() && top.onRegister())
+				compiler.releaseRegisterToStack(top)
+				compiler.returnFunction()
+
+				// Assemble.
+				code, _, err := compiler.compile()
+				require.NoError(t, err)
+				// Run code.
+				eng := newEngine()
+				mem := newMemoryInst()
+				jitcall(
+					uintptr(unsafe.Pointer(&code[0])),
+					uintptr(unsafe.Pointer(eng)),
+					uintptr(unsafe.Pointer(&mem.Buffer[0])),
+				)
+				// Check the stack.
+				require.Equal(t, uint64(1), eng.stackPointer)
+				if tc.signed {
+					require.Equal(t, tc.x1 >= tc.x2, eng.stack[eng.stackPointer-1] == 1)
 				} else {
-					require.Equal(t, uint64(0), eng.stack[eng.stackPointer-1])
+					require.Equal(t, uint64(tc.x1) >= uint64(tc.x2), eng.stack[eng.stackPointer-1] == 1)
 				}
 			})
 		}
 	})
+	t.Run("float32", func(t *testing.T) {
+		for i, tc := range []struct {
+			x1, x2 float32
+		}{
+			{x1: 100, x2: -1.1},
+			{x1: -1, x2: 100},
+			{x1: 100, x2: 200},
+			{x1: 100.01234124, x2: 100.01234124},
+			{x1: 100.01234124, x2: -100.01234124},
+			{x1: 200.12315, x2: 100},
+			{x1: float32(math.Inf(1)), x2: 100},
+			{x1: float32(math.Inf(-1)), x2: 100},
+		} {
+			t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+				// Prepare operands.
+				compiler := requireNewCompiler(t)
+				compiler.initializeReservedRegisters()
+				err := compiler.compileConstF32(&wazeroir.OperationConstF32{Value: tc.x1})
+				require.NoError(t, err)
+				x1 := compiler.locationStack.peek()
+				err = compiler.compileConstF32(&wazeroir.OperationConstF32{Value: tc.x2})
+				require.NoError(t, err)
+				x2 := compiler.locationStack.peek()
+
+				// Emit the Ge instructions,
+				err = compiler.compileGe(&wazeroir.OperationGe{Type: wazeroir.SignedTypeFloat32})
+				require.NoError(t, err)
+
+				// At this point, these registers must be consumed.
+				require.NotContains(t, compiler.locationStack.usedRegisters, x1.register)
+				require.NotContains(t, compiler.locationStack.usedRegisters, x2.register)
+				// Plus the result must be pushed.
+				require.Equal(t, uint64(1), compiler.locationStack.sp)
+
+				// To verify the behavior, we push the flag value
+				// to the stack.
+				flag := compiler.locationStack.peek()
+				require.True(t, flag.onConditionalRegister() && !flag.onRegister())
+				err = compiler.moveConditionalToGeneralPurposeRegister(flag)
+				require.NoError(t, err)
+				require.True(t, !flag.onConditionalRegister() && flag.onRegister())
+				compiler.releaseRegisterToStack(flag)
+				compiler.returnFunction()
+
+				// Assemble.
```

Comment:
wordy, but why not since here :D

```suggestion
				// Generate the code under test (constants declaration and comparison)
				// and the verification code (moving the result to the stack so we can assert against it)
```